lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -143,8 +143,9 @@ tokenize(String, Line, Column, Opts) ->
 tokenize(String, Line, Opts) ->
   tokenize(String, Line, 1, Opts).
 
-tokenize([], _Line, _Column, #elixir_tokenizer{terminators=[]}, Tokens) ->
-  {ok, lists:reverse(Tokens)};
+tokenize([], _Line, _Column, #elixir_tokenizer{terminators=[], warnings = Warnings}, Tokens) ->
+  UniqueWarnings = sets:to_list(sets:from_list(Warnings)),
+  {ok, {lists:reverse(Tokens), UniqueWarnings}};
```

Comment:
I don't think we need to make them unique, they are not meant to repeat. Let's also return a three element tuple:

```suggestion
  {ok, lists:reverse(Tokens), Warnings};
```

Reply:
Actually, now that I think about it, we don't need to return the warnings, we can just emit them here. So you can do something like this:

```suggestion
  [elixir_errors:erl_warn(Line, File, Msg) || {Line, File, Msg} <- lists:reverse(Warnings)],
  {ok, lists:reverse(Tokens)};
```

The reverse is necessary if you implement the `prepend_warnings` function I mentioned below.